sw/device/lib/testing/BUILD

```diff
@@ -0,0 +1,30 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+package(default_visibility = ["//visibility:public"])
+
+load("//rules:opentitan.bzl", "OPENTITAN_CPU")
+
+cc_library(
+    name = "testing",
```

Comment:
I think it's important to see what this target is doing. Is there a reason we don't have separate targets for check, coverage, and status? Why is this target defined here and not in `test_framework`?

Reply:
This is going to have to evolve and we're going to have to figure out how to integrate coverage measurements with `bazel coverage`.

IMHO, its ok for now, but leave behind a TODO to fix this up later.

Reply:
There's a tradeoff here between being concise and verbose. I've created this issue to track issues that might arise from implicitly depending on unneccesary targets because of the one collective and generally used testing target here. https://github.com/lowRISC/opentitan/issues/9098

Reply:
I strongly suspect the reason it's defined here is just an accident of Meson being terrible. I genuinely would not ascribe too much meaning to it.